src/Hazelcast.Net.Linq.Async/Visitors/QueryFormatter.cs

```diff
@@ -0,0 +1,501 @@
+﻿// Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Globalization;
+using System.Linq.Expressions;
+using System.Text;
+using Hazelcast.Core;
+using Hazelcast.Linq.Expressions;
+using Ionic.Zip;
+
+namespace Hazelcast.Linq.Visitors
+{
+    /// <summary>
+    /// Traverse and translate tree to SQL string statement.
+    /// </summary>
+    internal class QueryFormatter : HzExpressionVisitor
+    {
+        private StringBuilder _sb;
+        private List<object> _values;
+
+        // internal for test
+        internal QueryFormatter()
+        {
+            _sb = new();
+            _values = new();
+        }
+
+        /// <summary>
+        /// Translate the tree into SQL statement
+        /// </summary>
+        /// <param name="expression">tree</param>
+        /// <returns>(SQL statement, Value of variables)</returns>
+        public static (string, IEnumerable<object>) Format(Expression expression)
+        {
+            var f = new QueryFormatter();
+            f.Visit(expression);
+            return (f.ToString(), f._values);
+        }
+
+        public override string ToString()
+        {
+            return _sb.ToString();
+        }
+
+#pragma warning disable CS8765 // Nullability of type of parameter doesn't match overridden member (possibly because of nullability attributes).
```

Comment:
is this due to a conflict with the implemented interface not being nullable-enabled? like... if we implement nullable in our file, and the interface is not enabled, we have that type of conflicts?

Reply:
ok